packages/components/src/components/NewModal/NewModal.tsx

```diff
@@ -0,0 +1,208 @@
+import { ReactNode } from 'react';
+import styled from 'styled-components';
+import { useEvent } from 'react-use';
+import { borders, Elevation, mapElevationToBackground, spacingsPx } from '@trezor/theme';
+
+import { IconButton } from '../buttons/IconButton/IconButton';
+import { Text } from '../typography/Text/Text';
+import { H3 } from '../typography/Heading/Heading';
+import { ElevationContext } from '../ElevationContext/ElevationContext';
+import { useScrollShadow } from '../../utils/useScrollShadow';
+import { NewModalButton } from './NewModalButton';
+import { NewModalContext } from './NewModalContext';
+import { ModalAlignment, NewModalBackdrop } from './NewModalBackdrop';
+import { NewModalProvider } from './NewModalProvider';
+import type { NewModalVariant } from './types';
+import { Icon, IconType } from '../Icon/Icon';
+import { mapVariantToIconBackground, mapVariantToIconBorderColor } from './utils';
+import type { UISize } from '../../config/types';
+
+const NEW_MODAL_CONTENT_ID = 'modal-content';
+const MODAL_ELEVATION = 0;
+const ICON_SIZE = 40;
+
+type NewModalSize = Extract<UISize, 'large' | 'medium' | 'small'>;
+
+const getSize = (size: NewModalSize) => {
+    switch (size) {
```

Comment:
nit: we usually use mapping functions, but this is also fine

Reply:
Something like mapSizeToWidth?